### PR TITLE
Geocoding consolidation

### PR DESCRIPTION
**Pull request by mgraber**
First attempt at geocoding consolidation.
+ Single geocoding function that calls 1B, 1B-tpad, 2, 3, depending on whether or not stretch/streetname parsing.
+ Remove geocoding and address parsing from recipes, instead consolidating in geo.py
+ As much as possible, standardize the workflow used to call address cleaning prior to geocoding.
+ For recipes that have geo_rejects tables in EDM, include the process to identify records without geometries and move them to the geo_rejects table.

### CODE REVIEW

**Review by SPTKL — APPROVED** [review]

**Review by SPTKL — APPROVED** [review]

File: recipes/_helper/geo.py [hunk 1]
```diff
@@ -59,7 +78,7 @@ def clean_house(s):
     Returns:
     s (str): Cleaned house number
     """
-    s = ' ' if s == None else s
+    s = ' ' if s is None else s
```
Comment (AmandaDoyle) [inline recipes/_helper/geo.py:81]: Why is this ' ' instead of ''?

File: recipes/_helper/geo.py [hunk 2]
```diff
@@ -133,110 +166,68 @@ def air_geocode(inputs):
     geo (dict): Contains inputs as well as geosupport fields
         See geo_parser() for full list of fields
     """
-    hnum, sname, borough, street_name_1, street_name_2 = (
-        str("" if inputs[k] is None else inputs[k])
-        for k in ("hnum", "sname", "borough", "streetname_1", "streetname_2")
+    hnum, sname, borough, street_name_1, street_name_2, street_name_3 = (
```
Comment (AmandaDoyle) [inline recipes/_helper/geo.py:169]: Where does `street_name_1` and the like get set?  I'm following `street_1`but got lost at `street_name_1`.

File: recipes/dep_cats_permits/create.sql [hunk 3]
```diff
@@ -34,14 +34,14 @@ CREATE TEMP TABLE dep_cats_permits (
 
 \COPY dep_cats_permits FROM PSTDIN DELIMITER '|' CSV HEADER;
```
Comment (AmandaDoyle) [inline recipes/dep_cats_permits/create.sql:58]: Could benefit from comment to summarize what's going on below

File: recipes/dep_cats_permits/build.py [hunk 4]
```diff
@@ -4,60 +4,10 @@
 import pandas as pd
 import numpy as np
 import re
-from _helper.geo import get_hnum, get_sname, air_geocode as geocode
+from _helper.geo import get_hnum, get_sname, clean_street, clean_house, clean_boro_name, \
```
Comment (AmandaDoyle) [inline recipes/dep_cats_permits/build.py:7]: Could benefit from comment describing what's generally going on below

**Review by AmandaDoyle — COMMENTED** [review]
Looks good.  Once we talk through my two comments and I understand it fully we can approve and merge this in

**Pull request merged**